src/main/java/org/mskcc/cmo/metadb/service/impl/UpdateMessageHandlingServiceImpl.java

```diff
@@ -0,0 +1,275 @@
+            sampleFilterPhaser.register();
+            exec.execute(new SampleUpdateFilterHandler(sampleFilterPhaser));
+        }
+        sampleFilterPhaser.arriveAndAwaitAdvance();
+    }
+
+    private class RequestUpdateFilterHandler implements Runnable {
+        final Phaser phaser;
+        boolean interrupted = false;
+
+        RequestUpdateFilterHandler(Phaser phaser) {
+            this.phaser = phaser;
+        }
+
+        @Override
+        public void run() {
+            phaser.arrive();
+            while (true) {
+                try {
+                    String requestJson = requestUpdateFilterQueue.poll(100, TimeUnit.MILLISECONDS);
+                    if (requestJson != null) {
+                        Boolean passCheck = validRequestChecker.isValidRequestMetadataJson(requestJson);
+
+                        if (passCheck) {
+                            LOG.info("Sanity check passed, publishing to: " + SERVER_REQUEST_UPDATE_TOPIC);
+                            messagingGateway.publish(
+                                    SERVER_REQUEST_UPDATE_TOPIC,
+                                    requestJson);
+                        } else {
+                            LOG.error("Sanity check failed on request updates: "
+                                    + validRequestChecker.getRequestId(requestJson));
+                            requestStatusLogger.logRequestStatus(requestJson,
+                                    RequestStatusLogger.StatusType.REQUEST_UPDATE_FAILED_SANITY_CHECK);
+                        }
+                    }
+                    if (interrupted && requestUpdateFilterQueue.isEmpty()) {
+                        break;
+                    }
+                } catch (InterruptedException e) {
+                    interrupted = true;
+                } catch (Exception e) {
+                    LOG.error("Error during request handling", e);
+                }
+            }
+            requestUpdateFilterHandlerShutdownLatch.countDown();
+        }
+    }
+
+    private class SampleUpdateFilterHandler implements Runnable {
+        final Phaser phaser;
+        boolean interrupted = false;
+
+        SampleUpdateFilterHandler(Phaser phaser) {
+            this.phaser = phaser;
+        }
+
+        @Override
+        public void run() {
+            phaser.arrive();
+            while (true) {
+                try {
+                    String sampleJson = sampleUpdateFilterQueue.poll(100, TimeUnit.MILLISECONDS);
+                    if (sampleJson != null) {
+                        Map<String, String> sampleMap = mapper.readValue(sampleJson, Map.class);
+                        Boolean isCmoSample = validRequestChecker.isCmo(sampleJson);
+                        Boolean hasRequestId = validRequestChecker.hasRequestId(sampleJson);
+
+                        if (isCmoSample) {
+                            if (validRequestChecker.isValidCmoSample(sampleMap, isCmoSample, hasRequestId)) {
```

Comment:
I'm wondering why this method `validRequestChecker. isValidCmoSample(...)` needs the param `isCmoSample` if we are only calling it for the `isCmoSample == true`? 

seems like either (1) we don't need to split the logic of valid cmo vs. non-cmo if it's going to use the parameter anyway or (2) if we keep the param then use a method called `validRequestChecker.isValidSample(...)` which _does_ use the value of `isCmoSample`